### ISSUE TITLE
Semaphore channels don't need to be initialized

### ISSUE THREAD

**Issue by ianlewis** [RichardKnop/machinery#735]
Semaphore channels can be push style rather than pull.

For example:
https://github.com/RichardKnop/machinery/blob/7ccba3d50e4edd1489bd0ab1743534920deb093b/v2/brokers/redis/goredis.go#L97-L102

Can be implemented like:
```golang
pool := make(chan struct{}, concurrency)

select{
case pool <- struct{}{}:
    go func() {
        defer func() { <- pool }
        // Do concurrent thing.
    }()
}
```

**Comment by oskov** [comment]
Also in both redis brokers (goredis.go / redis.go has similar code) this pool is useless as it's always used by only one goroutine. Not sure what the original idea was, maybe multiple goroutines should simultaneous fetch data from redis or maybe this pool should be shared with the b.consume method. But definitely something is missing.

**Comment by ianlewis** [comment]
@oskov Yeah, concurrency is kind of ambiguously implemented. See #733 for my thoughts.

Some work probably needs be done to make this stuff cleaner.
